meson: Builds blkzone despite missing linux/blkzoned.h header

The Autotools project for util-linux checks for the existence of the `linux/blkzoned.h` to determine whether or not to build the `blkzone` executable. Unfortunately, this check is not carried over to Meson. This results in the following error when `linux/blkzoned.h` is not present:

```
[203/284] Compiling C object blkzone.p/sys-utils_blkzone.c.o
FAILED: blkzone.p/sys-utils_blkzone.c.o 
gcc -Iblkzone.p -I. -I../src -Iinclude -I../src/include -Ilibblkid -I../src/libblkid -I../src/libblkid/src -Ilibsmartcols -I../src/libsmartcols -I../src/libsmartcols/src -Ilibmount -I../src/libmount -I../src/libmount/src -Ilibfdisk -I../src/libfdisk -I../src/libfdisk/src -I../src/libuuid/src -I../src/liblastlog2/src -Isys-utils -I../src/sys-utils -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O3 -D_GNU_SOURCE -include /home/conan/workspace/prod-v2/bsr/4858/ceeac/p/b/libmo41aebcf197a14/b/build-release/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -MD -MQ blkzone.p/sys-utils_blkzone.c.o -MF blkzone.p/sys-utils_blkzone.c.o.d -o blkzone.p/sys-utils_blkzone.c.o -c ../src/sys-utils/blkzone.c
../src/sys-utils/blkzone.c:37:10: fatal error: linux/blkzoned.h: No such file or directory
   37 | #include <linux/blkzoned.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

This effects 2.40-rc2.

--- a/meson.build
+++ b/meson.build
@@ -1561,26 +1561,30 @@ exes += exe
 manadocs += ['sys-utils/blkdiscard.8.adoc']
 bashcompletions += ['blkdiscard']
 
-exe = executable(
-  'blkzone',
-  blkzone_sources,
-  include_directories : includes,
-  link_with : [lib_common],
-  install_dir : sbindir,
-  install : true)
-exes += exe
-manadocs += ['sys-utils/blkzone.8.adoc']
-bashcompletions += ['blkzone']
+if cc.has_header('linux/blkzoned.h')
+  exe = executable(
+    'blkzone',
+    blkzone_sources,
+    include_directories : includes,
+    link_with : [lib_common],
+    install_dir : sbindir,
+    install : true)
+  exes += exe
+  manadocs += ['sys-utils/blkzone.8.adoc']
+  bashcompletions += ['blkzone']
+endif
 
-exe = executable(
-  'blkpr',
-  blkpr_sources,
-  include_directories : includes,
-  link_with : [lib_common],
-  install_dir : sbindir,
-  install : true)
-exes += exe
-manadocs += ['sys-utils/blkpr.8.adoc']
+if cc.has_header('linux/pr.h')
+  exe = executable(
+    'blkpr',
+    blkpr_sources,
+    include_directories : includes,
+    link_with : [lib_common],
+    install_dir : sbindir,
+    install : true)
+  exes += exe
+  manadocs += ['sys-utils/blkpr.8.adoc']
+endif
 
 exe = executable(
   'ldattach',

